jmespath-jakarta-jsonp/src/main/java/io/burt/jmespath/jakarta/jsonp/JsonpRuntime.java

```diff
@@ -0,0 +1,230 @@
+    if (str.getValueType() == STRING) {
+      return ((JsonString)str).getString();
+    } else {
+      return str.toString();
+    }
+  }
+
+  @Override
+  public Number toNumber(JsonValue n) {
+    return (n.getValueType() == NUMBER) ? ((JsonNumber)n).numberValue() : null;
+  }
+
+  @Override
+  public boolean isTruthy(JsonValue value) {
+    switch (value.getValueType()) {
+      case FALSE:
+      case NULL:
+        return false;
+      case NUMBER:
+      case TRUE:
+        return true;
+      case ARRAY:
+        return ((JsonArray)value).size() > 0;
+      case OBJECT:
+        return ((JsonObject)value).size() > 0;
+      case STRING:
+        return ((JsonString)value).getString().length() > 0;
+      default:
+        throw new IllegalStateException(String.format("Unknown node type encountered: %s", value.getValueType()));
+    }
+  }
+
+  @Override
+  public JmesPathType typeOf(JsonValue value) {
+    switch (value.getValueType()) {
+      case ARRAY:
+        return JmesPathType.ARRAY;
+      case OBJECT:
+        return JmesPathType.OBJECT;
+      case STRING:
+        return JmesPathType.STRING;
+      case FALSE:
+      case TRUE:
+        return JmesPathType.BOOLEAN;
+      case NULL:
+        return JmesPathType.NULL;
+      case NUMBER:
+        return JmesPathType.NUMBER;
+      default:
+        throw new IllegalStateException(String.format("Unknown node type encountered: %s", value.getValueType()));
+    }
+  }
+
+  @Override
+  public JsonValue getProperty(JsonValue value, JsonValue name) {
+    if (value.getValueType() == OBJECT) {
+      return nodeOrNullNode(((JsonObject)value).get(textOnStringOrNull(name)));
+    } else {
+      return JsonValue.NULL;
+    }
+  }
+
+  @Override
+  public Collection<JsonValue> getPropertyNames(JsonValue value) {
+    if (value.getValueType() == OBJECT) {
+      Set<String> nameSet = ((JsonObject)value).keySet();
+      List<JsonValue> names = new ArrayList<>(nameSet.size());
+      for(String n : nameSet) {
+        names.add(createString(n));
+      }
+      return names;
+    } else {
+      return Collections.emptyList();
+    }
+  }
+
+  @Override
+  public JsonValue createNull() {
+    return nodeOrNullNode(null);
+  }
+
+  @Override
+  public JsonValue createArray(Collection<JsonValue> elements) {
+    JsonArrayBuilder builder = Json.createArrayBuilder();
+    for(JsonValue element : elements) {
+      builder.add(useJsonNull(element));
+    }
+    return builder.build();
+  }
+
+  @Override
+  public JsonValue createString(String str) {
+    return useJsonNull(Json.createValue(str));
+  }
+
+  @Override
+  public JsonValue createBoolean(boolean b) {
+    return b ? JsonValue.TRUE : JsonValue.FALSE;
+  }
+
+  @Override
+  public JsonValue createObject(Map<JsonValue, JsonValue> obj) {
+    JsonObjectBuilder builder = Json.createObjectBuilder();
+    for (Map.Entry<JsonValue, JsonValue> entry : obj.entrySet()) {
+      String key = textOnStringOrNull(entry.getKey());
```

Comment:
Can `key` really be null here? That's as far as I know not allowed in JSON, and the docs for the `createObject` method in `Adapter` says "The map keys must all be string values."

The Jackson adapter does not check for null in its implementation of this method so it's safe to skip the check.